be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -136,6 +136,11 @@ class SegmentIterator final : public ChunkIterator {
             bool may_has_del_row = chunk->delete_state() != DEL_NOT_SATISFIED;
             for (size_t i = 0; i < _column_iterators.size(); i++) {
                 const ColumnPtr& col = chunk->get_column_by_index(i);
+                if (_prune_column_after_index_filter && _prune_cols.count(i)) {
+                    // make sure each pruned column has the same size as the unpruneable one.
+                    col->resize(range.span_size());
```

Comment:
why not directly remove the prune column from context chunk?

Reply:
I think it is a more safety way to implement this optimization. Remove the column in chunk need to change more code in _build_context which maybe a little aggressive and more risky